### DISCUSSION TITLE
Should fenced and indented markdown codeblocks be parsing differently?

### DISCUSSION THREAD

**Discussion by solomon-b** [jgm/pandoc#10473] [Q&A] [answered]
When using fenced codeblocks I'm seeing newline characters get stripped from my codeblocks and the codeblock is wrapped in a `Para` node. Indented codeblocks preserve the newline and don't wrap in a `Para`. Is there a way to make fenced codeblocks behave the same as indented ones?

#### Fenced
Example codeblock:
````
```
id :: forall a. a -> a
id x  = x
```
````

Pandoc (json serialized) AST:
```
{
    "blocks": [
      {
        "c": [
          {
            "c": [
              [
                "",
                [],
                []
              ],
              "id :: forall a. a -> a id x  = x"
            ],
            "t": "Code"
          }
        ],
        "t": "Para"
      }
    ],
    "meta": {},
    "pandoc-api-version": [
      1,
      23,
      1
    ]
  }
```

#### Space Indented 
Example codebloack:
```
    id :: forall a. a -> a
    id x  = x
```

Pandoc (json serialized) AST:
```
{
    "blocks": [
      {
        "c": [
          [
            "",
            [],
            []
          ],
          "id :: forall a. a -> a\nid x  = x"
        ],
        "t": "CodeBlock"
      }
    ],
    "meta": {},
    "pandoc-api-version": [
      1,
      23,
      1
    ]
  }

```

**Comment by kysko** [comment]
Gives correct output for me (pandoc 3.6 Windows).

Only way I could reproduce your first output was using this as input:

```
`
id :: forall a. a -> a
id x  = x
`
```
ie using only one tick, which would be a Para with an inline Code.

**Comment by jgm** [comment]
I can't reproduce what you're reporting.  Use https://pandoc.org/try and get a permalink if you can reproduce there.

**Comment by solomon-b** [comment]
I'm using pandoc as a library. This is the exact snippet I used to generate the code:

```
  pandocResult <- liftIO . runIO $ do
    let readerConfig = def { readerExtensions = extensionsFromList [Ext_fenced_code_blocks, Ext_hard_line_breaks] }
        writerConfig = def { writerExtensions =  extensionsFromList [Ext_fenced_code_blocks, Ext_hard_line_breaks] }
    pandocAst <- readMarkdown readerConfig md
    result <- writeHtml5String writerConfig pandocAst
    pure (pandocAst, result)
```

Here is the input `Text`:
```
"```\nid :: forall a => a -> a\nid x = x\n```\n\n    id :: forall a => a -> a\n    id x = x"
```
and here is the Pandoc AST
```
Pandoc (Meta {unMeta = fromList []}) [Para [Code ("",[],[]) "id :: forall a => a -> a id x = x"],CodeBlock ("",[],[]) "id :: forall a => a -> a\nid x = x"]
```

This is with `pandoc-3.5`. I'll try 3.6 and report back.

EDIT: Same results with `pandoc-3.6`

**Comment by jgm** [comment, marked as answer]
Check the documentations for the extensions you're using:
https://pandoc.org/MANUAL.html#extension-fenced_code_blocks
The `fenced_code_blocks` extension is for tilde-fenced code blocks (a pandoc innovation that predates GFM style fenced code blocks). You want `backtick_code_blocks`.

**Reply by solomon-b** [reply]
Thank you that solved it!

**Reply by solomon-b** [reply]
Where can I see what extensions are enabled in the pandoc binary?

**Reply by jgm** [reply]
See T.P.Extensions in the code.
And `--list-extensions` in the manual.
